tests/test_resample.py

```diff
@@ -0,0 +1,35 @@
+import numpy as np
+
+from examples.seismic import TimeAxis, RickerSource, demo_model
+
+
+def test_resample():
+
+    shape = (50, 50, 50)
+    spacing = (10., 10., 10.)
+    nbpml = 10
+
+    f0 = 0.01
+    t0 = 0.0
+    nt = 10000
+
+    # Create two-layer model from preset
+    model = demo_model(preset='layers-isotropic', vp_top=1., vp_bottom=2.,
+                       spacing=spacing, shape=shape, nbpml=nbpml)
+
+    time_range = TimeAxis(t0, model.critical_dt, nt)
+    src_a = RickerSource(name='src_a', grid=model.grid, f0=f0, time_range=time_range)
+
+    time_range_f = TimeAxis(t0, None, 10*nt, stop=time_range.stop)
+    src_b = RickerSource(name='src_b', grid=model.grid, f0=f0, time_range=time_range_f)
+
+    src_c = src_b.resample(dt=model.critical_dt)
+
+    assert src_a.data.shape == src_c.data.shape
+
+    res = src_a.data - src_c.data
+    assert np.linalg.norm(res) < 0.005
```

Comment:
Relative error are better usually for comparison:
`np.isclose( np.linalg.norm(res)/np.linalg.norm(src_a.data), 1.0, rtol=1e-6)`